msal/src/Microsoft.Identity.Client/UIParent.cs

```diff
@@ -116,20 +118,24 @@ public static bool IsSystemWebviewAvailable()
 
             int counter = 0;
 
-            ApplicationInfo applicationInfo = Application.Context.PackageManager.GetApplicationInfo(_chromePackages[0], 0);
-
-            for (int i = 0; i < _chromePackages.Length; i++)
+            ApplicationInfo applicationInfo = Application.Context.PackageManager.GetApplicationInfo(_customTabPackages[0], 0);
+            List<string> customTabPackages = new List<string>();
+            for (int i = 0; i < _customTabPackages.Length; i++)
             {
                 try
                 {
-                    packageManager.GetPackageInfo(_chromePackages[i], PackageInfoFlags.Activities);
+                    customTabPackages.Add(packageManager.GetPackageInfo(_customTabPackages[i], PackageInfoFlags.Activities).ToString());
                     counter++;
                 }
                 catch (PackageManager.NameNotFoundException)
                 {
                 }
             }
-            if (counter > 0 && applicationInfo.Enabled == true)
+            if (counter == 1 && !string.IsNullOrEmpty(customTabPackages[0]) && !applicationInfo.Enabled)
```

Comment:
>counter [](start = 16, length = 7)

I think you can simplify this logic if you pull out the chrome stuff separtely, maybe not have any counter variable at all.

Reply:
i tried this. let me know what you think. 

---
In reply to: [227737484](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1347#discussion_r227737484) [](ancestors = 227737484)